valhalla/baldr/graphtile.h

```diff
@@ -315,7 +315,7 @@ class GraphTile {
 
   // Graph tile memory, this must be shared so that we can put it into cache
   // Apparently you can std::move a non-copyable
-  boost::shared_array<char> graphtile_;
+  boost::shared_ptr<std::vector<char>> graphtile_;
```

Comment:
having it as a `vector` maybe we dont even need it to be a `shared_ptr` we can essentially just change anything that does:

    if(graphtile_)

into:

    if(graphtile_.size())

what do you think?